### PR TITLE
New Home Page

### LINKED ISSUE #6056: Add toggle for switching between new homepage and legacy homepage
Add toggle in advanced setting for switching on and off for the new homepage.

### PR DESCRIPTION
**Pull request by abbyhu2000**
### Description
This PR is built upon the original home page PR: https://github.com/opensearch-project/OpenSearch-Dashboards/pull/5613 authored by @BSFishy 

Added a new commit on replacing the yml config with advanced setting.

### Issues Resolved
resolves #6056 

## Screenshot

Screenshot 2024-03-07 at 11 42 16 AM

### Check List

- [ ] All tests pass
  - [ ] `yarn test:jest`
  - [ ] `yarn test:jest_integration`
- [ ] New functionality includes testing.
- [ ] New functionality has been documented.
- [ ] Update [CHANGELOG.md](./../CHANGELOG.md)
- [ ] Commits are signed per the DCO using --signoff

### CODE REVIEW

**Comment by abbyhu2000** [comment]
Currently i placed the advanced setting under general section. Do we need to change any wording here? @kgcreative @ashwin-pc

File: src/core/server/ui_settings/settings/home.ts
Comment (ashwin-pc) [inline src/core/server/ui_settings/settings/home.ts:1]: Cant we add this setting to a new section in the home plugin? Similar to how discover does it? https://github.com/suzhou-joe/opensearch-dashboards/blob/b2f79ca6dfce478be51e5bd394d1676bd75e46ca/src/plugins/discover/server/ui_settings.ts#L58

Comment (abbyhu2000) [reply src/core/server/ui_settings/settings/home.ts:1]: yes, i just moved the location of the uisettings within the home plugin.

File: .github/workflows/cypress_workflow.yml [hunk 1]
```diff
@@ -28,7 +28,7 @@ on:
 
 env:
   TEST_REPO: ${{ inputs.test_repo != '' && inputs.test_repo || 'opensearch-project/opensearch-dashboards-functional-test' }}
-  TEST_BRANCH: "${{ inputs.test_branch != '' && inputs.test_branch || github.base_ref }}"
+  TEST_BRANCH: "${{ 'feature/new-home' || inputs.test_branch != '' && inputs.test_branch || github.base_ref }}"
```
Comment (AMoo-Miki) [inline .github/workflows/cypress_workflow.yml:31]: Do we need this?

**Comment by AMoo-Miki** [comment]
I would recommend using the changelog from `main` and adding the entry. The one here has too many changes and almost all are undesired.

File: .github/workflows/cypress_workflow.yml (hunk 1 above)
Comment (ashwin-pc) [reply .github/workflows/cypress_workflow.yml:31]: nope

**Review by ashwin-pc — COMMENTED** [review]
LGTM except for Miki's suggestion and the changelog conflict. Will approve once thats in.

**Comment by kgcreative** [comment]
Cc: @lauralexis

**Comment by abbyhu2000** [comment]
Should add some fallback mechanism if getting the new home page failed. https://github.com/opensearch-project/OpenSearch-Dashboards/issues/6146

**Review by ananzh — APPROVED** [review]

**Comment by ananzh** [comment]
Thanks for the great work @BSFishy @abbyhu2000

**Pull request merged**